include/miopen/config.h.in

```diff
@@ -78,12 +78,55 @@
 
 // clang-format off
 #cmakedefine MIOPEN_DEFAULT_FIND_MODE @MIOPEN_DEFAULT_FIND_MODE@
+
 // clang-format on
+#ifndef MIOPEN_CMP_FLAGS_NEW
 #cmakedefine MIOPEN_AMDGCN_ASSEMBLER "@MIOPEN_AMDGCN_ASSEMBLER@"
 #cmakedefine HIP_OC_COMPILER "@HIP_OC_COMPILER@"
 #cmakedefine MIOPEN_HIP_COMPILER "@MIOPEN_HIP_COMPILER@"
 #cmakedefine MIOPEN_OFFLOADBUNDLER_BIN "@MIOPEN_OFFLOADBUNDLER_BIN@"
 #cmakedefine MIOPEN_CACHE_DIR "@MIOPEN_CACHE_DIR@"
+#else /*MIOPEN_CMP_FLAGS_NEW if defined*/
+/* Include rocm-core header for get ROCm Install Path Method */
+#include <rocm-core/rocm_getpath.h>
+
+/* Flags to help identify/differentiate Definitions for each MIOPEN Compiler Flags */
+typedef enum {
+        MIOPEN_AMDGCN_Flag=0,
+        MIOPEN_OC_COMPILER_Flag=1,
+        MIOPEN_HIP_COMPILER_Flag=2,
+        MIOPEN_OFFLOAD_BUNDLER_Flag=3,
+        MIOPEN_ROCM_INSTALL_PATH_Flag=4,
+        MIOPEN_CMP_MAX_Flag=5 /*Should be last Flag definition*/
+} MIOPEN_CMP_Flags_t;
+
+/* List to hold Relative Directory Path for each Compiler Flags from ROCM Install Path */
+extern const char* MIOPEN_CMP_FLAG_REL_PATHS[MIOPEN_CMP_MAX_Flag];
```

Comment:
Declaring it const says that you promise not to change it under program control. Do not lie to the compiler!